service/service.go

```diff
@@ -483,12 +509,132 @@ func (s *service) BeforeServe(
+
 	if _, ok := csictx.LookupEnv(ctx, "X_CSI_VXFLEXOS_NO_PROBE_ON_START"); !ok {
 		return s.doProbe(ctx)
 	}
 	return nil
 }
 
+func (s *service) updateConfigMap(getIPAddressByInterfacefunc GetIPAddressByInterfacefunc, configFilePath string) {
+	configFileData, err := os.ReadFile(configFilePath)
+	if err != nil {
+		Log.Errorf("Failed to read ConfigMap file: %v", err)
+		return
+	}
+
+	var config Config
+	err = yaml.Unmarshal(configFileData, &config)
+	if err != nil {
+		Log.Errorf("Failed to parse configMap data: %v", err)
+		return
+	}
+
+	updateInterfaceNamesWithIPs := map[string]string{}
+	for node, interfaceList := range config.InterfaceNames {
+
+		if !strings.EqualFold(node, s.opts.KubeNodeName) {
+			continue
+		}
+		interfaces := strings.Split(interfaceList, ",")
+		var ipAddresses []string
+
+		for _, interfaceName := range interfaces {
+			interfaceName = strings.TrimSpace(interfaceName)
+
+			// Find the IP of the Interfaces
+			ipAddress, err := getIPAddressByInterfacefunc(interfaceName, &service{})
+			if err != nil {
+				Log.Printf("Error while getting IP address for interface %s: %v\n", interfaceName, err)
+				continue
+			}
+			ipAddresses = append(ipAddresses, ipAddress)
+		}
+		if len(ipAddresses) > 0 {
+			updateInterfaceNamesWithIPs[node] = strings.Join(ipAddresses, ",")
+		}
+	}
+
+	// Get the Kubernetes ClientSet
+	if K8sClientset == nil {
+		err = k8sutils.CreateKubeClientSet()
+		if err != nil {
+			Log.Errorf("Failed to create Kubernetes ClientSet: %v", err)
+			return
+		}
+		K8sClientset = k8sutils.Clientset
+	}
+
+	// Get the vxflexos-config-params ConfigMap
+	cm, err := K8sClientset.CoreV1().ConfigMaps(DriverNamespace).Get(context.TODO(), DriverConfigMap, metav1.GetOptions{})
+	if err != nil {
+		Log.Errorf("Failed to get ConfigMap: %v", err)
+		return
+	}
+
+	var configData map[string]interface{}
+	if existingYaml, ok := cm.Data["driver-config-params.yaml"]; ok {
+
+		err := yaml.Unmarshal([]byte(existingYaml), &configData)
+		if err != nil {
+			Log.Errorf("Failed to parse ConfigMap data: %v", err)
+			return
+		}
+
+		// Check and update Interfaces with the IPs
+		if interfaceNames, ok := configData["interfaceNames"].(map[string]interface{}); ok {
+			for node, ipAddressList := range updateInterfaceNamesWithIPs {
+				interfaceNames[node] = ipAddressList
+			}
+		} else {
+			Log.Errorf("interfaceNames key missing or not in expected format")
+			return
+		}
+
+		updatedYaml, err := yaml.Marshal(configData)
+		if err != nil {
+			Log.Errorf("Failed to marshal updated data: %v", err)
+			return
+		}
+		cm.Data["driver-config-params.yaml"] = string(updatedYaml)
+	}
+
+	// Update the vxflexos-config-params ConfigMap
+	_, err = K8sClientset.CoreV1().ConfigMaps("vxflexos").Update(context.TODO(), cm, metav1.UpdateOptions{})
+	if err != nil {
+		Log.Errorf("Failed to update ConfigMap: %v", err)
+		return
+	}
+
+	fmt.Println("ConfigMap updated successfully")
```

Comment:
Use Log instead of fmt.Println for this message

Reply:
done